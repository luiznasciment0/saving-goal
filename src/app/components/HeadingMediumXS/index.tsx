import Typography from 'app/components/Typography'

interface HeadingMediumXSProps {
  children: React.ReactNode
}

const HeadingMediumXS = ({ children }: HeadingMediumXSProps) => (
  <Typography
    as="h2"
    size="xxlarge"
    weight="normal"
    color="secondaryBlue"
    family="secondary"
  >
    {children}
  </Typography>
)

export default HeadingMediumXS
