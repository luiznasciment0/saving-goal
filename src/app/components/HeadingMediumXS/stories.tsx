import { Story, Meta } from '@storybook/react/types-6-0'
import HeadingMediumXS from '.'

export default {
  title: 'HeadingMediumXS',
  component: HeadingMediumXS,
  argTypes: {
    children: {
      type: 'string'
    }
  }
} as Meta

export const Default: Story = (args) => (
  <HeadingMediumXS>{args.children}</HeadingMediumXS>
)

Default.args = {
  children: 'Teste'
}
