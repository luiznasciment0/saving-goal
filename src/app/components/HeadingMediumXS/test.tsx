import { render, screen } from '@testing-library/react'

import HeadingMediumXS from '.'

describe('<HeadingMediumXS />', () => {
  it('should render the HeadingMediumXS and its styles', () => {
    const { container } = render(<HeadingMediumXS>Test</HeadingMediumXS>)

    expect(screen.getByRole('heading', { name: /test/i })).toHaveStyle({
      'font-size': '2.4rem',
      'line-height': '2.88rem',
      'font-weight': '500',
      'font-family': `'Rubik'`,
      color: '#0079ff'
    })

    expect(container.firstChild).toMatchSnapshot()
  })
})
