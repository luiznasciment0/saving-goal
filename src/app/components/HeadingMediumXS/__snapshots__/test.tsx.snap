// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[
  `<HeadingMediumXS /> should render the HeadingMediumXS and its styles 1`
] = `
.c0 {
  font-size: 2.4rem;
  line-height: 2.88rem;
  font-family: 'Rubik';
  color: #0079ff;
  font-weight: 500;
}

<h2
  class="c0"
  color="secondaryBlue"
>
  Teste
</h2>
`
